raft/src/test/java/org/apache/kafka/raft/internals/VoterSetHistoryTest.java

```diff
@@ -89,6 +89,27 @@ void testAddAt() {
         assertEquals(Optional.empty(), votersHistory.valueAtOrBefore(99));
         assertEquals(Optional.of(addedVoterSet), votersHistory.valueAtOrBefore(199));
         assertEquals(Optional.of(removedVoterSet), votersHistory.valueAtOrBefore(200));
+
+        // Assert multiple voters can be added or removed at a time
+        voterMap.put(4, VoterSetTest.voterNode(4, true));
+        voterMap.put(5, VoterSetTest.voterNode(5, true));
+
+        VoterSet addedMultipleVoters = VoterSet.fromMap(new HashMap<>(voterMap));
+        votersHistory.addAt(300, addedMultipleVoters);
+
+        assertEquals(addedMultipleVoters, votersHistory.lastValue());
+        assertEquals(Optional.of(removedVoterSet), votersHistory.valueAtOrBefore(299));
+        assertEquals(Optional.of(addedMultipleVoters), votersHistory.valueAtOrBefore(300));
+
+        voterMap.remove(4);
+        voterMap.remove(5);
+
+        VoterSet removedMultipleVoters = VoterSet.fromMap(new HashMap<>(voterMap));
+        votersHistory.addAt(400, removedMultipleVoters);
+
+        assertEquals(removedMultipleVoters, votersHistory.lastValue());
+        assertEquals(Optional.of(addedMultipleVoters), votersHistory.valueAtOrBefore(399));
+        assertEquals(Optional.of(removedMultipleVoters), votersHistory.valueAtOrBefore(400));
     }
```

Comment:
The test looks good but let's put it in a separate test that is checking this case specifically. This test is already getting hard to read.